workers/datasets_based/src/datasets_based/workers/first_rows.py

```diff
@@ -526,24 +570,18 @@ def compute_first_rows_response(
             "Server error while post-processing the split rows. Please report the issue.",
             cause=err,
         ) from err
-    # get the size of the surrounding JSON (without the rows)
-    features_list = to_features_list(features=features)
-    response: FirstRowsResponse = {
-        "dataset": dataset,
-        "config": config,
-        "split": split,
-        "features": features_list,
-        "rows": [],
-    }
-    surrounding_json_size = get_json_size(response)
+
     # truncate the rows to fit within the restrictions, and prepare them as RowItems
+    surrounding_json_size = get_json_size(response)
     row_items = create_truncated_row_items(
         rows=transformed_rows,
         min_cell_bytes=min_cell_bytes,
         rows_max_bytes=rows_max_bytes - surrounding_json_size,
         rows_min_number=rows_min_number,
     )
     response["rows"] = row_items
+    validate_content_size(response, rows_max_bytes=rows_max_bytes)
```

Comment:
I think it's not necessary here. In the worst case, `create_truncated_row_items` should return an empty array, which would give the same size as surrounding_json_size.

Reply:
But we have to change a bit the implementation of `create_truncated_row_items` to effectively return an empty array (if I remember correctly, I forced having at least one row)

Reply:
@severo  I reviewed the code and couldn't find where the function  `create_truncated_row_items` returns at least one, but anyway I think we should add that validation (_If size is bigger return empty array)_ in the `compute_first_rows_response` function instead on adding that logic in `create_truncated_row_items` since its only responsibility should be only "truncate" rows but not "reduce" the response, that could be  part of the compute logic, what do you think?

Reply:
Sorry, I just realized that logic is managed into truncate_row_item and features+rows are combined which makes difficult to manage outside in compute_first_rows_response. I removed the extra validation since it is no more needed and handled in another function

Reply:
> But we have to change a bit the implementation of create_truncated_row_items to effectively return an empty array (if I remember correctly, I forced having at least one row)

Sorry I was wrong... Indeed: we keep at least `min_rows_number` rows, and if needed, we truncate their content until the size is under `max_rows_bytes - surrounding_json_size`. If the parameters `FIRST_ROWS_MAX_BYTES`, `FIRST_ROWS_MIN_CELL_BYTES`, `FIRST_ROWS_MIN_NUMBER` and `FIRST_ROWS_COLUMNS_MAX_NUMBER` are set correctly, truncating should get the size under the threshold. It's not guaranteed because the features, even if their number is under `FIRST_ROWS_COLUMNS_MAX_NUMBER`, can have any size (think of nested dictionaries), but I don't think we should over-engineer here. It would work for most cases.

Maybe a least we should add a check here: https://github.com/huggingface/datasets-server/blob/main/workers/datasets_based/src/datasets_based/workers/first_rows.py#L267. If, after truncating all the cells of the 10 (`min_rows_number`) first rows, the size is still over the maximum, raise a new kind of error. It would avoid an UnknownError.